### PR TITLE
feat: Query for data

### PR DESCRIPTION
**Pull request by mure**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nisystemlink-clients-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- Adds `query_table_data` method for advanced table filtering and sorting
- Adds `query_decimated_data` for decimated queries

### What testing has been done?

- Added automated tests for the new methods

### CODE REVIEW

File: tests/integration/dataframe/test_dataframe.py [hunk 1]
```diff
@@ -6,18 +6,30 @@
...
-
-basic_table_model = models.CreateTableRequest(
-    columns=[
-        models.Column(
-            name="index",
-            data_type=models.DataType.Int32,
-            column_type=models.ColumnType.Index,
-        )
-    ]
+from nisystemlink.clients.dataframe.models import (
```
Comment (mure) [inline tests/integration/dataframe/test_dataframe.py:9]: The `models` namespace is making the tests unnecessarily verbose, so I converted it to individual imports.

**Review by cameronwaterman — APPROVED** [review]

File: nisystemlink/clients/dataframe/_data_frame_client.py [hunk 2]
```diff
@@ -197,3 +197,15 @@ def append_table_data(self, id: str, data: models.AppendTableDataRequest) -> Non
...
+    @post("tables/{id}/query-data", args=[Path, Body])
+    def query_table_data(
+        self, id: str, query: models.QueryTableDataRequest
+    ) -> models.PagedTableRows:
+        """Reads rows of data that match a filter from the table identified by its ID.
+
+        Args:
+            id: Unique ID of a DataFrame table.
+            query: The filtering and sorting to apply when reading data.
+        """
```
Comment (mure) [inline nisystemlink/clients/dataframe/_data_frame_client.py:210]: I forgot to add the return type again. It's in my current change set for decimated queries.

File: tests/integration/dataframe/test_dataframe.py [hunk 3]
```diff
@@ -340,10 +341,134 @@ def test__write_invalid_data__raises(
...
+
+        client.append_table_data(
+            id, AppendTableDataRequest(frame=frame, end_of_data=True)
+        )
+
+        # TODO: Remove mock when service supports flushing
+        with responses.RequestsMock() as rsps:
+            rsps.add(
+                responses.POST,
+                f"{client.session.base_url}tables/{id}/query-data",
```
Comment (spanglerco) [inline tests/integration/dataframe/test_dataframe.py:375]: Are we able to verify the request JSON being sent? These tests aren't actually testing the parameters being passed to the API, just that they can deserialize the response JSON.

**Review by spanglerco — CHANGES_REQUESTED** [review]

File: tests/integration/dataframe/test_dataframe.py (hunk 3 above)
Comment (mure) [reply tests/integration/dataframe/test_dataframe.py:380]: Turns out the `responses` library makes it really easy to match JSON bodies 🥳

**Review by cameronwaterman — APPROVED** [review]

File: nisystemlink/clients/dataframe/models/_query_decimated_data_request.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    specified method is ``MAX_MIN`` or ``ENTRY_EXIT``. Only numeric columns are
+    supported. i.e. ``INT32``, ``INT64``, ``FLOAT32``, ``FLOAT64`` and
+    ``TIMESTAMP``."""
+
+    intervals: Optional[int] = None
+    """Number of intervals to use for decimation."""
+
+    method: Optional[DecimationMethod] = None
+    """Specifies the method used to decimate the data"""
+
```
Comment (spanglerco) [inline nisystemlink/clients/dataframe/models/_query_decimated_data_request.py:48]: Should we document that the defaults are 1000 intervals and LOSSY decimation? Same for the `decimation` property of `QueryDecimatedDataRequest` below (if you leave out the options, you'll get LOSSY decimation with 1000 intervals over the Index column).

**Review by spanglerco — APPROVED** [review]

File: nisystemlink/clients/dataframe/models/_query_decimated_data_request.py (hunk 4 above)
Comment (mure) [reply nisystemlink/clients/dataframe/models/_query_decimated_data_request.py]: Good call. I documented the defaults.

**Pull request merged**